src/transformers/generation/utils.py

```diff
@@ -1421,6 +1424,11 @@ def generate(
                 )
             generation_config.max_length = generation_config.max_new_tokens + input_ids_length
 
+        # adjust max_length when using `input_embeds` in decoder-only models
```

Comment:
Rather than saying what this is doing (we can tell from the code) it would be useful for the comment to explain why we need to do this.